### ISSUE TITLE
Road::GetWidth() doesn't work for side = 0

### ISSUE THREAD

**Issue by brifsttar** [esmini/esmini#96]
Hi Emil,

First of all: don't worry, it's not a big problem! :-D

`road::GetWidth`'s doc mention that we can give `side=0` to get both side's width.

https://github.com/esmini/esmini/blob/2a90268fbb563b9819fc14c979deb077e5a8d04b/EnvironmentSimulator/Modules/RoadManager/RoadManager.hpp#L971

However, it seems that's not the case, as the method only checks `side > 0`

https://github.com/esmini/esmini/blob/2a90268fbb563b9819fc14c979deb077e5a8d04b/EnvironmentSimulator/Modules/RoadManager/RoadManager.cpp#L1706-L1716

So passing `side = 0` as parameter gives the same result as `side = -1`.

I'm not sure if you'd prefer to simply remove this `0` case from the documentation, or actually implement it. In any case, it's definitely not a headache-inducing issue ;-)

On a side note, we've been using RoadManager in more and more use-cases, and it's working perfectly fine, with no issue to report. So once again, thank you for your amazing work.

Bertrand

**Comment by eknabevcc** [comment]
this was a very handy problem for a change 😋
Thanks a lot for reporting. A clear bug - I must have missed that special, but important, case in some refactorization of the function.
Now it should be fixed (by commit [46f7fb1](https://github.com/esmini/esmini/commit/46f7fb103182b850f7b4f4b3e8da70434bc81243)). And also added a [unit test](https://github.com/esmini/esmini/blob/46f7fb103182b850f7b4f4b3e8da70434bc81243/EnvironmentSimulator/Unittest/RoadManager_test.cpp#L1448) to catch any regression.

**Comment by brifsttar** [comment]
Hi Emil,

Thanks for the quick saturday answer :-)

However your fix doesn't properly solve the issue. If you still have my OpenDRIVE file, you can check out road 2185: it's a ~10m one-way road (i.e. one side only) for which `GetWidth` returns ~20m.

I think the root cause is you forgot to reset the `lane_id` in case `side == 0`

https://github.com/esmini/esmini/blob/f3ec101bd9658cc0bef77fa631fc10f78837d605/EnvironmentSimulator/Modules/RoadManager/RoadManager.cpp#L1725-L1733

This means that the same `lane_id` is used for both calls to `GetOuterOffset`, so `offset1` simply gets the same value as `offset0`, which actually passes your test because your road has the same width on both sides.

You might want to add also this line after resetting `index`, 
https://github.com/esmini/esmini/blob/f3ec101bd9658cc0bef77fa631fc10f78837d605/EnvironmentSimulator/Modules/RoadManager/RoadManager.cpp#L1715

I did this change locally and it works. I could do a PR for that, but I think it'd be easier for the both of us if I don't ;-)

Bertrand

**Comment by eknabevcc** [comment]
Damn it. I knew I should have found an OpenDRIVE with non symmetrical road. But I was lazy, which not always is not a good strategy :-)
Thanks for quick test and reporting. Of course, you're right, it is a clear bug. 
I think pull request is great for several reasons, one is to automatically see the true contributor. But we can try that next time. Let's do it simple this time. I've fixed according to your proposal. 👍
I also changed the test to involve non symmetric road (I found an example in soderleden.xodr). And added a variant of the test which measure only drivable part.
Commit: https://github.com/esmini/esmini/commit/9ddf3cba09173681cf7885454cdec1a25c74c2a7
